### PR TITLE
Stand-alone point in polyhedron test

### PR DESCRIPTION
**Pull request by OmarDuran**
## Proposed changes

The purpose of this PR is to remove the external implementation for point-in-polyhedron test. An immediate benefit is a cleaner installation of PorePy. The proposed implementation is supported by computing a generalized winding number documented here (https://dl.acm.org/doi/10.1145/2461912.2461916). The implementation is robust and supports non-convex polyhedra.

 * The documentation was adapted
 * Some undesired prints were removed from `test_mpfa_gravity.py` and `test_tag.py`

## Types of changes

What types of changes does your code introduce to PorePy?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing (contribution related mainly to testing of existing functionality)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation 
- [x] I have checked that I have not duplicated existing functionality

### CODE REVIEW

File: src/porepy/geometry/point_in_polyhedron_test.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        point P. It is the measure of the intersection of the three-dimensional unit
+        sphere and the conic hull defined by the vectors in R.
+
+        The implementation is based on equation (6) in reference: Robust inside-outside
+            segmentation using generalized winding numbers
+            (https://doi.org/10.1145/2461912.2461916)
+
+        Args:
+            R (np.ndarray, num_pt x 3): Triangle points relocated to the origin point
+                [0,0,0]. Since the unit sphere is centered at P, we translated the
```
Comment (IvarStefansson) [inline src/porepy/geometry/point_in_polyhedron_test.py:47]: The past tense is a bit confusing. Does this mean that the method should only be called for translated triangles?

File: src/porepy/geometry/point_in_polyhedron_test.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        """
+
+        r = np.linalg.norm(R, axis=1)
+
+        check_overlapping_vertices = np.any(np.absolute(r) < self.tol)
+        if check_overlapping_vertices:
+            raise ValueError("Origin point coincides with a vertex")
+
+        # Check collinearity by computing areas formed by pair of points in R and
```
Comment (IvarStefansson) [inline src/porepy/geometry/point_in_polyhedron_test.py:66]:
```suggestion
        # Check collinearity by computing areas formed by pairs of points in R and
```

**Review by IvarStefansson — COMMENTED** [review]
Looks good to me, but I think @keileg should also review this one.

File: src/porepy/geometry/point_in_polyhedron_test.py (hunk 1 above)
Comment (OmarDuran) [reply src/porepy/geometry/point_in_polyhedron_test.py]: Agree. I have rewritten those lines.

**Comment by keileg** [comment]
I'll have a look early next week, please do not merge before then.

File: src/porepy/geometry/point_in_polyhedron_test.py [hunk 3]
```diff
@@ -36,16 +36,16 @@ def solid_angle(self, R: np.ndarray[Any, np.dtype[np.float64]]) -> float:
...
             segmentation using generalized winding numbers
             (https://doi.org/10.1145/2461912.2461916)
 
         Args:
-            R (np.ndarray, num_pt x 3): Triangle points relocated to the origin point
-                [0,0,0]. Since the unit sphere is centered at P, we translated the
-                original triangle's points by substracting the point being tested
+            R (np.ndarray, num_pt x 3): Translated triangle's points at origin (0,0,0).
+            The original triangle's points need to be translated by subtracting the
+            arbitrary point P
```
Comment (IvarStefansson) [inline src/porepy/geometry/point_in_polyhedron_test.py:48]:
```suggestion
            arbitrary point P.
```

File: src/porepy/geometry/point_in_polyhedron_test.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+
+
+class PointInPolyhedronTest(object):
```
Comment (keileg) [inline src/porepy/geometry/point_in_polyhedron_test.py:8]:
```suggestion
class PointInPolyhedronTest:
```
No need to inherit from object in python 3.

File: src/porepy/geometry/point_in_polyhedron_test.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    The implementation is based on the reference: Robust inside-outside segmentation
+        using generalized winding numbers (https://doi.org/10.1145/2461912.2461916)
+
+    Implementation requires consistent orientation of the triangulated surface.
+
+    Attributes:
+        vertices (np.ndarray, num_pt x 3): Triangulation vertices.
+        connectivity (np.ndarray num_triangles x 3): Triangulation connectivity map.
+        tol (float, optional): Geometric tolerance, used in comparison of
+            points, areas and volumes. Defaults to 1e-10.
```
Comment (keileg) [inline src/porepy/geometry/point_in_polyhedron_test.py:21]:
```suggestion
            points, areas and volumes. Defaults to 1e-10.

```

File: src/porepy/geometry/point_in_polyhedron_test.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        Args:
+            R (np.ndarray, num_pt x 3): Translated triangle's points at origin (0,0,0).
+            The original triangle's points need to be translated by subtracting the
+            arbitrary point P.
+
+        Raises:
+            ValueError: If the origin [0,0,0] point:
+                coincides with a vertex
+                is collinear with the vertices
+                is coplanar with the vertices
```
Comment (keileg) [inline src/porepy/geometry/point_in_polyhedron_test.py:54]:
```suggestion
                is coplanar with the vertices

```

**Review by keileg — APPROVED** [review]
A few suggestions on formating. Good work.

**Pull request merged**